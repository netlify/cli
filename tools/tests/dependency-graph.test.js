const { DependencyGraph } = require('../project-graph')

/** @type {DependencyGraph} */
let graph

beforeEach(() => {
  graph = new DependencyGraph()
  graph.addDependency('tests/a.js', 'src/nested/a.js')
  graph.addDependency('tests/c.js', 'src/c/index.js')
  graph.addDependency('tests/c.js', 'tests/utils.js')
  graph.addDependency('src/nested/a.js', 'src/nested/b.js')
  graph.addDependency('src/nested/a.js', 'src/c/index.js')
  graph.addDependency('src/c/index.js', 'src/d.js')
})

test('should test if all parents are affected by changing a src file on the bottom', () => {
  expect(graph.affected(['src/d.js'])).toMatchSnapshot()
})

test('should test only the root leaf is affected if the root one is passed', () => {
  expect(graph.affected(['tests/a.js'])).toMatchSnapshot()
})

test('should test that nothing is affected if the passed file is not in the dependency graph', () => {
  expect([...graph.affected(['some-markdown.md'])]).toHaveLength(0)
})
