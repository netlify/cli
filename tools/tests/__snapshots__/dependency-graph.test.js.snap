// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test if all parents are affected by changing a src file on the bottom 1`] = `
Set {
  "src/d.js",
  "src/c/index.js",
  "src/nested/a.js",
  "tests/a.js",
  "tests/c.js",
}
`;

exports[`should test only the root leaf is affected if the root one is passed 1`] = `
Set {
  "tests/a.js",
}
`;
